src/camlsnark_c/libsnark-caml/libsnark/caml/caml_curve.cpp.template

```diff
@@ -477,6 +481,40 @@ const linear_combination<FieldT>* CURVE_PREFIX(r1cs_constraint_c)(const r1cs_con
   return &c->c;
 }
 
+typedef struct CURVE_PREFIX(r1cs_evaluations) {
+  std::vector<FieldT>* a;
+  std::vector<FieldT>* b;
+  std::vector<FieldT>* c;
+} r1cs_evaluations;
+
+r1cs_evaluations CURVE_PREFIX(r1cs_constraint_system_evaluations)(
+    const r1cs_constraint_system<FieldT>* sys,
+    const std::vector<FieldT>* full_variable_assignment,
+    const size_t degree)
+{
+    const size_t d_plus_1 = degree + 1;
+    std::vector<FieldT>* ca = new std::vector<FieldT>(d_plus_1, FieldT::zero());
+    std::vector<FieldT>* cb = new std::vector<FieldT>(d_plus_1, FieldT::zero());
+    std::vector<FieldT>* cc = new std::vector<FieldT>(d_plus_1, FieldT::zero());
+
+    for (size_t i = 0; i <= sys->num_inputs(); ++i)
+    {
+        ca->at(i + sys->num_constraints()) = (i > 0 ? full_variable_assignment->at(i-1) : FieldT::one());
```

Comment:
This seems like an unusual place to keep the variable assignments.. Is there a reason that we can't use the `full_variable_assignment` vector directly (and thus also remove the zero-padding at the end of `cb` and `cc`)?